### PR TITLE
feat(release-please): add default token input option

### LINKED ISSUE #211: The token can set the default input
The token can set the `${{ secrets.GITHUB_TOKEN }}` as the default input.

The `${{ github.token }}` can access to it.

### PR DESCRIPTION
**Pull request by yi-Xu-0100**
set the ${{ secrets.GITHUB_TOKEN }} as the default input of token.

The ${{ github.token }} can access to it.

Close #211

### CODE REVIEW

**Comment by yi-Xu-0100** [comment]
@googlebot I signed it!

**Comment by pragmaticivan** [comment]
It might be worth adding an example of using a custom token instead of GITHUB_TOKEN as it's still a valid input.

**Comment by yi-Xu-0100** [comment]
@pragmaticivan I tried to add an example usage for the personal access token. 😃

**Comment by yi-Xu-0100** [comment]
Hey @bcoe, I have been using this feature (use default token for action) in my actions for a long time. e.g., [yi-Xu-0100/repo-list-generator](https://github.com/yi-Xu-0100/repo-list-generator) and [yi-Xu-0100/traffic-to-badge](https://github.com/yi-Xu-0100/traffic-to-badge). What else do I need to do to merge this PR?

**Comment by bcoe** [comment]
@yi-Xu-0100 I'm slightly surprised by this feature, because it means that an upstream action can get the GITHUB_TOKEN, without it explicitly being set as inputs.

Is there a step where an action needs to opt in? this is interesting from a security perspective for actions.

**Comment by yi-Xu-0100** [comment]
@bcoe 

`github.token` used as the default input value of the token in the official actions([checkout](https://github.com/actions/checkout/blob/25a956c84d5dd820d28caab9f86b8d183aeeff3d/action.yml#L24), [stale](https://github.com/actions/stale/blob/07f3f88b6d7554724b296fec903b4635a085f4dc/action.yml#L8), [github-script](https://github.com/actions/github-script/blob/b251c3e139ca75460518e90ba9800f97235e9ae7/action.yml#L13)) is also very common. I think that the reasons are as follows:
1. The permission of `secret.GITHUB_TOKEN` is very limitation, and it is restricted to be used in workflows, and does not have the permission to obtain special data in the repository or the data in other repositories of the owner. (Reference [permissions-for-the-github_token](https://docs.github.com/en/actions/reference/authentication-in-a-workflow#permissions-for-the-github_token))
2. There are steps that must use token in the current action. Token is a required item, and `secret.GITHUB_TOKEN` with the least authority is used as the default input, which may reduce repeated configuration.
3. The `github.token` can be used to obtain the same permissions as `secret.GITHUB_TOKEN`. (Reference [github-context](https://docs.github.com/en/actions/reference/context-and-expression-syntax-for-github-actions#github-context))

File: README.md [hunk 1]
```diff
@@ -36,7 +35,7 @@ Automate releases with Conventional Commit Messages.
 
 | input | description |
 |:---:|---|
-| `token` | A GitHub secret token, you will most likely want to use the special `secrets.GITHUB_TOKEN` |
+| `token` | A GitHub secret token, the action default use the special `secrets.GITHUB_TOKEN` |
```
Comment (bcoe) [inline README.md:38]:
```suggestion
| `token` | A GitHub secret token, the action defaults to using the special `secrets.GITHUB_TOKEN` |
```

**Review by bcoe — COMMENTED** [review]
@yi-Xu-0100 will review tomorrow I'm hoping 👍

File: README.md [hunk 2]
```diff
@@ -215,6 +209,38 @@ jobs:
           git push origin v${{ steps.release.outputs.major }}.${{ steps.release.outputs.minor }}
 ```
 
+### Use the personal access token in fork repository
```
Comment (bcoe) [inline README.md:212]: I think we could just drop this section. It's a lot of content, and I think it's clear from the description of `--token` that you can provide an alternate token if you're interested.

**Review by bcoe — CHANGES_REQUESTED** [review]
Looks good to me, I think we could do without the additional section in the README. What do you think?

**Comment by bcoe** [comment]
@yi-Xu-0100 @pragmaticivan if we want to keep the section about a custom token, I might just make it a short blurb, about the fact that you can use an alternate token, perhaps pointing out the `fork` option.

I don't think we need a fully fleshed out example, folks will understand they can just pass an alternate token.

**Comment by yi-Xu-0100** [comment]
@bcoe I tried to reduce the example section, hoping it could be better. 😀

**Comment by yi-Xu-0100** [comment]
@bcoe When I test for my own repository, I find the token of `secret.GITHUB_TOKEN` can be used for both the fork and private repository. 

At first, I test the fork repository, and it does well. See in https://github.com/yi-xu-test-organization/test-repo-1/pull/1 (the input value of fork is true because of the boolean input issue)

Then I set the origin repository (https://github.com/yi-Xu-0100/test-repo) was setting to be private to test the action and also got the success result.

So I delete the example section and only keep the description of the token.  😅

**Review by bcoe — APPROVED** [review]

**Pull request merged**